### PR TITLE
Translation Documentation 

### LINKED ISSUE #681: Update developer documentation to reflect i18n changes

### LINKED ISSUE #682: Update author documentation to reflect i18n changes

### PR DESCRIPTION
**Pull request by jordangriffiths01**
Fixes #681 
Fixes #682

### CODE REVIEW

File: docs/source/developer/translation.rst [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+The basic idea is that for each translatable field, an extra database column is added for every language defined in the django settings file.
+When the base field is accessed, ``modeltranslation`` performs some magic to retrieve the translation for the users language
+
+
+To register translation fields for a given TranslatableModel, add a TranslationOptions subclass to the ``translations.py`` file in the relevant application.
+For more details, see the models already registered, or read the `modeltranslation docs <http://django-modeltranslation.readthedocs.io/en/latest/registration.html>`_.
+
+.. note::
+
+  The default behaviour for modeltranslation is to fallback to the english value if no translation is present.
```
Comment (JackMorganNZ) [inline docs/source/developer/translation.rst]: 'modeltranslation' should be within code tags, and 'english' to 'English'

File: docs/source/developer/translation.rst [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+  In CS Unplugged, this is desirable for fields such as name and title, but is often undesirable for most other fields (see `Translation Principles`_).
```
Comment (JackMorganNZ) [inline docs/source/developer/translation.rst]: 'this is desirable for fields' to 'this is desirable for text fields'

File: docs/source/author/translations.rst [hunk 3]
```diff
@@ -1,5 +1,97 @@
...
+  Any roadblocks encountered and lessons learned will be used to refine the translation workflow and improve this guide.
+
+Getting Started
+==============================================================================
+
+It's really easy to start contributing translations using CS Unplugged
+
+1. Read this documentation page, especially the `Translation Notes`_ section.
+2. Create an account on `Crowdin <https://crowdin.com/join>`__.
+3. Read the Crowdin `documentation <https://support.crowdin.com/crowdin-intro/>`_ for translators.
```
[the remark this answers is not in the document]
Comment (jordangriffiths01) [reply docs/source/author/translations.rst]: Fair enough - I actually meant to link to https://support.crowdin.com/online-editor/ which is more relevant. But that's listed below anyway, so I've removed this step.

File: docs/source/author/translations.rst [hunk 4]
```diff
@@ -1,5 +1,97 @@
...
+4. Visit the CS Unplugged `development site <http://cs-unplugged-dev.appspot.com/>`_.
+5. Switch to translation mode using the language drop down.
+6. Log in with your Crowdin credentials and select a target language.
+7. Start translating!
+
+Crowdin Overview
+==============================================================================
+We use a localisation management platform called `Crowdin <https://crowdin.com/project/cs-unplugged>`__ for translation of CS Unplugged.
+Our project is public, meaning that anyone can create an account and contribute translations.
```
[the remark this answers is not in the document]
Comment (jordangriffiths01) [reply docs/source/author/translations.rst]: This is true, it got opened up a while back

File: docs/source/developer/translation.rst [hunk 5]
```diff
@@ -50,6 +52,11 @@ Both types of file are stored inside the directory tree for a given language (ie
 
   These files can be parsed and loaded using a `utility function <UtilityFunctions_>`_ on the ``TranslatableModelLoader`` base class.
 
+For the latter two type of files, Django's built-in `translation support <https://docs.djangoproject.com/en/2.0/topics/i18n/>`_ is utilised to handle translatable strings.
+In python code, text is wrapped in a ``ugettext`` function call (usually aliased to ``_``).
```
Comment (JackMorganNZ) [inline docs/source/developer/translation.rst]: 'python' to 'Python'

File: docs/source/author/topics.rst [hunk 6]
```diff
@@ -60,15 +64,34 @@ For more information about what this page displays, see :doc:`../developer/dev`.
...
+
+There are 3 different types of files used for adding content to CS Unplugged:
+
+- Content markdown files
+- YAML files containing translatable strings
+- YAML configuration files
+
+All files live inside the ``topics/content`` directory.
+The first two files are unique for each translation language, and are stored in a directory tree specific to that language.
+This directory is named using the languages django locale code eg. ``en``, ``de_AT``.
```
Comment (JackMorganNZ) [inline docs/source/author/topics.rst]: 'django' to 'Django'

File: docs/source/author/topics.rst [hunk 7]
```diff
@@ -60,15 +64,34 @@ For more information about what this page displays, see :doc:`../developer/dev`.
...
 ==============================================================================
 
-The diagram below is an example of the ``content/en/`` language directory for
-the project's topic application, where:
+The ``content`` directory for the topics application contains
 
-- Blue is directories.
-- Red is YAML configuration files (see :doc:`understanding_configuration_files`).
-- Green is Markdown text files.
+- a directory for each language in which content exists, named using the django locale code. This directory contains
```
Comment (JackMorganNZ) [inline docs/source/author/topics.rst]: 'django' to 'Django'

File: docs/source/author/topics.rst [hunk 8]
```diff
@@ -60,15 +64,34 @@ For more information about what this page displays, see :doc:`../developer/dev`.
...
 
-.. raw:: html
-  :file: ../_static/html_snippets/topics_content_directory_tree.html
+  - Content markdown files
```
Comment (JackMorganNZ) [inline docs/source/author/topics.rst]: 'markdown' to 'Markdown'

File: docs/source/author/topics.rst [hunk 9]
```diff
@@ -1,6 +1,10 @@
 Topic Content
 ##############################################################################
 
+.. warning::
+  All flowcharts in this guide need to be updated to reflect changes to the content directory structure.
+  This is tracked by by `isssue/749 <https://github.com/uccser/cs-unplugged/issues/749>`__.
```
Comment (JackMorganNZ) [inline docs/source/author/topics.rst]: Double 'by'

File: docs/source/author/topics.rst [hunk 10]
```diff
@@ -60,15 +64,34 @@ For more information about what this page displays, see :doc:`../developer/dev`.
...
+- YAML configuration files
+
+All files live inside the ``topics/content`` directory.
+The first two files are unique for each translation language, and are stored in a directory tree specific to that language.
+This directory is named using the languages django locale code eg. ``en``, ``de_AT``.
+Configuration files are shared amongst all languages, because the content structure is the same for all languages.
+These files live under a special ``structure`` directory.
+
+As a simple rule, structure files situated inside the ``structure`` directory contain **no** user facing text.
+Any user facing text lives in either a markdown file or a YAML translation file inside the locale specific directories.
```
Comment (JackMorganNZ) [inline docs/source/author/topics.rst]: 'markdown' to 'Markdown'

File: docs/source/author/topics.rst [hunk 11]
```diff
@@ -60,15 +64,34 @@ For more information about what this page displays, see :doc:`../developer/dev`.
...
+- YAML files containing translatable strings
+- YAML configuration files
+
+All files live inside the ``topics/content`` directory.
+The first two files are unique for each translation language, and are stored in a directory tree specific to that language.
+This directory is named using the languages django locale code eg. ``en``, ``de_AT``.
+Configuration files are shared amongst all languages, because the content structure is the same for all languages.
+These files live under a special ``structure`` directory.
+
+As a simple rule, structure files situated inside the ``structure`` directory contain **no** user facing text.
```
Comment (JackMorganNZ) [inline docs/source/author/topics.rst]: Optional: Would be simpler to phrase it as below?

> 'directory contain **no** text a website user will see.

**Pull request merged**